Dockerfile

```diff
@@ -11,11 +11,30 @@ RUN poetry install --no-interaction --no-ansi
 COPY . ./app
 WORKDIR /app
 
+# testing stage
+FROM base as test
+
+RUN apt-get update &&\
+  apt-get upgrade -y &&\
+  curl -sSL https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb -o chrome.deb &&\
+  apt-get install ./chrome.deb -y &&\
+  rm ./chrome.deb
+RUN LATEST=`curl -sSL https://chromedriver.storage.googleapis.com/LATEST_RELEASE` &&\
+ echo "Installing chromium webdriver version ${LATEST}" &&\
+ curl -sSL https://chromedriver.storage.googleapis.com/${LATEST}/chromedriver_linux64.zip -o chromedriver_linux64.zip &&\
+ apt-get install unzip -y &&\
+ unzip ./chromedriver_linux64.zip
+
+ENTRYPOINT ["poetry", "run", "pytest"]
+
+
+# local development stage
 FROM base as development
 EXPOSE 5000
 ENTRYPOINT ["./run_dev.sh"]
 
+# production build stage
 FROM base as production
-RUN pip install gunicorn
-EXPOSE 8000
-ENTRYPOINT ["./run_prod.sh"]
\ No newline at end of file
+RUN pip install gunicorn flask
+EXPOSE $PORT
+ENTRYPOINT ["gunicorn -b 0.0.0.0:$PORT 'app:create_app()'"]
```

Comment:
This is only needed for module 8 (deploying to Heroku)

But a few things:
- This should be `ENTRYPOINT gunicorn -b 0.0.0.0:$PORT 'app:create_app()'` in order to actually execute as a shell command and have the $PORT replaced by the environment variable's value
- $PORT is now not optional - either update your README's instructions for running the production image, or add a default value here
- `EXPOSE $PORT` doesn't do anything because it's not a shell command - it can't access environment variables.